driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -273,14 +295,20 @@ private void setFinalResult(
           && logger.isWarnEnabled()) {
         logServerWarnings(response.warnings);
       }
+
       callback.onSet(connection, response, info, statement, System.nanoTime() - startTime);
+
+      span.setStatus(StatusCode.OK);
+      span.end();
     } catch (Exception e) {
       callback.onException(
           connection,
           new DriverInternalError(
               "Unexpected exception while setting final result from " + response, e),
           System.nanoTime() - startTime, /*unused*/
           0);
+      span.setStatus(StatusCode.ERROR, e.toString());
```

Comment:
I believe that adding `span.recordException(e)` before it will make the span more useful (although maybe more spammy?)